spec/services/open_library_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe OpenLibrary do
+  before do
+    stub_request(:get, 'https://openlibrary.org/api/books?bibkeys=ISBN:0385472579&format=json&jscmd=data')
+        .to_return(
+            status: 200,
+            body: load_fixture('get_response_success'),
+            headers: { 'Content-Type': 'application/json' }
+        )
+
+    stub_request(:get, 'https://openlibrary.org/api/books?bibkeys=ISBN:0385472579abc&format=json&jscmd=data')
+        .to_return(
+            status: 200,
+            body: load_fixture('get_response_not_found'),
+            headers: { 'Content-Type': 'application/json' }
+        )
+  end
+
+  it 'gets a book from open library' do
+    response = OpenLibrary.new.find_by_isbn('0385472579')
+    expect(response).equal?(load_fixture('get_response_success'))
+  end
+
+  it 'gets a book not found error from open library' do
+    response = OpenLibrary.new.find_by_isbn('0385472579abc')
```

Comment:
podes hacer is_expected to_throw(OpenLibraryException)

Reply:
o `expect { OpenLibrary.new.find_by_isbn('0385472579')  }.to throw(OpenLibraryException)`

Reply:
Fixed.